### DISCUSSION TITLE
when using SPA with remix is it good for seo ?

### DISCUSSION THREAD

**Discussion by shop-fluencer** [remix-run/remix#5051] [Q&A] [answered]
title says

**Comment by machour** [comment, marked as answer]
Answer: yes.

PS: Dear @shop-fluencer, this is your third opened issue/discussion in a very short time.
Would you consider reading (our documentation)[https://remix.run/docs/en/v1] or join our [Discord server](https://rmx.as/discord) for a more chat-like experience ? 😃

**Reply by shop-fluencer** [reply]
Hi sorry but I have a lot of questions :D I join your discord server then 👍

**Comment by cliffordfajardo** [comment]
Not sure I understand the question, but it sounds like you're asking if a Remix SPA app is good for SEO?

**Fullstack Remix (SSR / SPA)**
Yes - when you request a webpage from the server, remix will server side render (SSR) your page first, which is good for SEO;
how it generally works with SSR: when the search engine bots/crawlers reach the URL, they'll get the entire content (HTML) of the page since there is no javascript parsing that needs occur to build the HTML of the request page


**React SPA app using React Router (no SSR)**
If you're not doing any server side rendering and doing client side rendering (CSR) only, SEO is still possible. 

But if you have the opportunity to SSR your app on initial page request, which happens automatically by default with fullstack remix, then google / search engine bots dont need to wait on the javascript of your app to execute to build the entire application DOM (HTML);


The first paragraph of these official google docs has more details
https://developers.google.com/search/docs/crawling-indexing/javascript/javascript-seo-basics

Visuals on CSR vs SSR 
https://javascriptpatterns.vercel.app/patterns/rendering-patterns/client-side-rendering



## References
Discord Thread in `#help`: https://discord.com/channels/770287896669978684/1062434073718566982
